head/head.go

```diff
@@ -150,6 +150,15 @@ func NewHead(ctx context.Context, options ...opts.Option) (*Head, chan Bootstrap
 		}
 		providerStore = hproviders.CombineProviders(providerStore, hproviders.AddProviderNotSupported(delegateProvider))
 	}
+	if cfg.StoreTheIndexAddr != "" {
+		log.Infof("will delegate to %v with timeout %v", cfg.StoreTheIndexAddr, cfg.DelegateTimeout)
+		stiProvider, err := hproviders.StoreTheIndexProvider(cfg.StoreTheIndexAddr, cfg.DelegateTimeout)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to instantiate delegation client (%w)", err)
+		}
+		providerStore = hproviders.CombineProviders(providerStore, hproviders.AddProviderNotSupported(stiProvider))
```

Comment:
IIUC this will try the caching provider store, which we expect to fail, and then it will enqueue an async DHT lookup. Those are expensive, will always fail, and will contend for resources (the work queue) with regular DHT queries...is there a way to avoid that?

Reply:
Are you saying you're concerned that all the requests that end up being fulfilled by the indexers will result in DHT queries that will likely fail and you're concerned about the load?

If so we have some options here depending on the semantics we want. One might be that if we make a "fallback provider" that instead of trying all the providers in parallel does them sequentially only if the previous ones fail. In this case we could then decide to only do a DHT lookup in the event the Datastore and Indexer systems returned no records.

Reply:
if there's a change in logic here it should be in a different PR in order to keep the scope of this one reasonable.

Reply:
Sounds good, but can we agree to not deploy this to the Hydras until this is fixed?

Reply:
This change is not making the current situation any worse right?

Do we have a consensus agreement for something better?

Reply:
Agreed with Gus. In practice merging this code will make it worse. The change I proposed should be pretty small though

Reply:
I think i wasn't clear earlier - what i meant by 'this change' was that this PR uses the same composition structure as the already merged delegated routing structure. I agree that spinning up load in this path is something we need to watch in case it leads to lots of failing dht queries, and that the change to the composition structure propose seems good.

* There isn't going to be substantial amount of upstream bitswap data that we expect loaded into store the index in the coming week. it would be useful for providers to begin testing the end-to-end flow though, so if the additional change is going to take more than this coming week, we should consider if we can get away without it temporarily.

* @guseggert  if you're able to make the proposed change, that would be great!